### PR TITLE
Adds peering example

### PR DESCRIPTION
**Pull request by amandakarina**
Hey folks,

This pull request adds peering example for step 4-projects.

Could you please, take a look?

Thanks!

### CODE REVIEW

**Comment by rjerrems** [comment]
FYI it looks like tests are failing due to peering occurring concurrently ```       Error: Error adding network peering: googleapi: Error 400: There is a peering operation in progress on the local or peer network. Try again later., badRequest
       
         on .terraform/modules/projects_bu1_dev.peering/modules/network-peering/main.tf line 22, in resource "google_compute_network_peering" "local_network_peering":
         22: resource "google_compute_network_peering" "local_network_peering" {
       
       
       
       Error: Error adding network peering: googleapi: Error 400: There is a peering operation in progress on the local or peer network. Try again later., badRequest
       
         on .terraform/modules/projects_bu1_prod.peering/modules/network-peering/main.tf line 22, in resource "google_compute_network_peering" "local_network_peering":
         22: resource "google_compute_network_peering" "local_network_peering" {```

File: 4-projects/business_unit_1/development/example_peering_project.tf [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+locals {
+  vpc_name      = "${local.env_code}-peering-base"
```
Comment (rjerrems) [inline 4-projects/business_unit_1/development/example_peering_project.tf:19]: Given vpc_name is not reused anywhere else, should this information not just be encoded in network_name?

File: 4-projects/business_unit_1/development/example_peering_project.tf [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  network_name  = "vpc-${local.vpc_name}"
+  env_code      = "d"
+  environment   = "development"
```
Comment (rjerrems) [inline 4-projects/business_unit_1/development/example_peering_project.tf:22]: Can we follow the convention we are using elsewhere, where these are hard coded. e.g

https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/4-projects/business_unit_1/development/example_base_shared_vpc_project.tf

These values are not reused, so it does not provide much value and can make the code harder to read

File: test/integration/projects/controls/gcp_projects.rb [hunk 3]
```diff
@@ -55,6 +61,12 @@
   'p' => { 'bu1' => prod_bu1_project_floating, 'bu2' => prod_bu2_project_floating }
 }
 
+peering_projects_id = {
```
Comment (bharathkkb) [inline test/integration/projects/controls/gcp_projects.rb:64]: nit: can we add a tests that asserts the peering. 
something with `gcloud compute networks peerings list` perhaps?

**Review by bharathkkb — APPROVED** [review]
LGTM and thanks for adding the tests @amandakarina 
/cc @rjerrems

File: test/integration/projects/controls/gcp_projects.rb (hunk 3 above)
Comment (amandakarina) [reply test/integration/projects/controls/gcp_projects.rb:64]: yeap! I'll add!

**Review by rjerrems — APPROVED** [review]
LGTM

**Pull request merged**